src/app/queries/fetchHelpers.ts

```diff
@@ -49,6 +51,41 @@ export const authorizedFetch = async <T>(
   }
 };
 
+export const authorizedK8sFetch = async <T>(
+  resourceType: KubeResource,
+  fetchContext: IFetchContext,
+  client: ClusterClient
+): Promise<T> => {
+  const { history, setSelfSignedCertUrl, checkExpiry } = fetchContext;
+
+  try {
+    const response = await client.list(resourceType);
+
+    if (response && response.data) {
+      return response.data;
+    } else {
+      throw response;
+    }
+  } catch (error) {
+    checkExpiry(error, history);
+
+    const isAxiosSelfSignedCertError = (err) => {
+      // HACK: Doing our best to determine whether or not the
+      // error was produced due to a self signed cert error.
+      // It's an extremely barren object.
```

Comment:
That's a detail, don't change it unless you agree with me.
It's difficult to figure out what's the link between a self signed cert error here and "Network Error".
Also not sure how the HACK header is relevant.

Reply:
The problem here is a limitation of the way the cert errors are reported in this `err` object. They are literally just a TypeError with the message "Network Error", so it is a hacky solution that just assumes any error that looks like that is a cert error.

However, I actually don't think that message is consistent across Firefox and Chrome, @ibolton336 we might need different logic here. Can we check if it is an instance of TypeError like we do for the other helper or is axios obfuscating that?

Reply:
Right @mturley This all stems from the self signed cert temporarily solution until we start using a different solution (server side requests & cert storage, etc. ).  @gildub If you have a better solution please let me know otherwise this is the best approach I can think of for handling the "Network error" message thrown from axios. We arent getting a TypeError from axios - it is completely different than what we receive from fetch. Axios is obfuscating it.

Reply:
Double checking, I was wrong about the different message in Firefox, at least on mac it's working as written in both FF and Chrome. Yeah, I'm thinking more and more we should be proxying all the API calls through Express, this issue would go away.